chemprop/cli/train.py

```diff
@@ -105,6 +110,12 @@ def add_train_args(parser: ArgumentParser) -> ArgumentParser:
     # )
 
     transfer_args = parser.add_argument_group("transfer learning args")
+    transfer_args.add_argument(
+        "--checkpoint",
+        type=Path,
+        nargs="+",
+        help="Location of checkpoint(s) or model file(s) to be loaded for overwriting weights. It can be a path to either a single pretrained model checkpoint (.ckpt) or single pretrained model file (.pt), a directory that contains these files, or a list of path(s) and directory(s). If a directory, will recursively search and predict on all found (.pt) models.",
+    )
     transfer_args.add_argument(
         "--model-frzn",
         help="Path to model checkpoint file to be loaded for overwriting and freezing weights. By default, all MPNN weights are frozen with this option.",
```

Comment:
if we have both `--checkpoint` and `--model-frzn` flags, it seems like the logically operate in the same space. That is, both of these flags correspond to a filepath containing model weights to load in, with the only difference that the latter freezes these weights. In this case, I think we should change the `--model-frzn` to be a boolean flag, `--freeze-encoder`, which does exactly that: freezes the loaded encoder weights.

Reply:
and FWIW, providing a directory that will be searched for `.pt` and `.ckpt` files seems like a bad idea to me. It seems significantly more idiomatic for a user to directly supply these paths?

Reply:
> if we have both `--checkpoint` and `--model-frzn` flags, it seems like the logically operate in the same space. That is, both of these flags correspond to a filepath containing model weights to load in, with the only difference that the latter freezes these weights. In this case, I think we should change the `--model-frzn` to be a boolean flag, `--freeze-encoder`, which does exactly that: freezes the loaded encoder weights.

I also thought about this during the implementation. I think it would be more straightforward to do it the way you suggested. 

> and FWIW, providing a directory that will be searched for `.pt` and `.ckpt` files seems like a bad idea to me. It seems significantly more idiomatic for a user to directly supply these paths?

If a directory is used as input for `--checkpoint`, only the `.pt` files would be found. The reasons for doing providing this capability are (1) if we trained a model on a split with ensemble models, we can just pass the path to the directory to find all those ensemble models instead of passing them explicitly, (2) we can also reuse the `find_models` function we used in `predict.py`, and (3) in v1, we had `--checkpoint_dir` so old users might want this functionality to be implemented in v2 as well.